service/server.py

```diff
@@ -67,21 +68,38 @@ def get_properties_for_address(address):
     return query.execute().hits
 
 
+def paginated_address_records(address_records, page_number):
+    start_index = (page_number-1)*int(SEARCH_RESULTS_PER_PAGE)
+    end_index = max(page_number*int(SEARCH_RESULTS_PER_PAGE), len(address_records)-1)
```

Comment:
I was wondering how your new tests passed with this line. It turns out that `...[start_index:end_index]` is smart enough not to go beyond the the array. This means we just need this: `end_index = page_number*int(SEARCH_RESULTS_PER_PAGE)`